fieldsolver/gridGlue.cpp

```diff
@@ -179,6 +190,210 @@ void feedMomentsIntoFsGrid(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>&
 
   MPI_Waitall(sendRequests.size(), sendRequests.data(), MPI_STATUSES_IGNORE);
 
+  // ---------------------------------------------------------------------------------
+  // Boxcar Smoothening
+  const int *mntDims= &momentsGrid.getLocalSize()[0];
+  std::array<Real,fsgrids::moments::N_MOMENTS> *Grid;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridL1;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridL2;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridR1;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridR2;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *swap;
+
+
+  /*Kernel Weights are fractional numbers 
+
+              std::array weights[i]
+    weight= ---------------------------
+                2*refLevel+1
+
+  and are calculated as shown above while iterating over each dimensions
+  The numerator is calculated filling the weights array
+  and the denominator depends only on the stencil width. 
+  */
+  float weightC, weightL1, weightL2, weightR1, weightR2;
+  int blurPasses = 3;
+  int refLevel;
+  std::array<int, 5> weights;
+  weights.fill(0);
+
+  for (int blurPass = 1; blurPass <= blurPasses; blurPass++)
+
+  {
+
+    #pragma omp parallel for collapse(3)
+    for (int kk = 0; kk < mntDims[2]; kk++)
+    {
+      for (int jj = 0; jj < mntDims[1]; jj++)
+      {
+        for (int ii = 0; ii < mntDims[0]; ii++)
+        {
+
+          // Dont filter if in max resolution region
+          refLevel = technicalGrid.get(ii, jj, kk)->RefLevel;
+          if (refLevel == 0)
+          {
+            continue;
+          }
+
+          // Calculate Numerator
+          for (int index = 0; index < refLevel; index++)
```

Comment:
The refinement level can be anything, 3 or more, and then this will overflow. See comment above too.